### ISSUE TITLE
regression in azurerm_role_assignment - 2.62.0 (from 2.61.0)

### ISSUE THREAD

**Issue by surlypants** [terraform-providers/terraform-provider-azurerm#12168]
state:
```bash
$ terragrunt state list 2>/dev/null | grep "module.virtual_machines.azurerm_role_assignment.store_access"
module.virtual_machines.azurerm_role_assignment.store_access[0]
module.virtual_machines.azurerm_role_assignment.store_access[1]
module.virtual_machines.azurerm_role_assignment.store_access[2]
module.virtual_machines.azurerm_role_assignment.store_access[3]
```

plan with 2.61.0:
```bash
$ terragrunt init -upgrade=true -reconfigure && terragrunt plan -lock=false 2>/dev/null | grep "module.virtual_machines.azurerm_role_assignment.store_access" | grep -v "Refreshing state"
```
nothing to do (expected)

plan with 2.62.0:
```bash
$ terragrunt init -upgrade=true -reconfigure && terragrunt plan -lock=false 2>/dev/null | grep "module.virtual_machines.azurerm_role_assignment.store_access" | grep -v "Refreshing state"
  # module.virtual_machines.azurerm_role_assignment.store_access[0] will be created
  # module.virtual_machines.azurerm_role_assignment.store_access[1] will be created
  # module.virtual_machines.azurerm_role_assignment.store_access[2] will be created
  # module.virtual_machines.azurerm_role_assignment.store_access[3] will be created
```
regression here. state is ignored.

I observe this seemingly specifically related to storage account IAM roles -- triggered in VM and VMSS modules. I am _not_ seeing this for IAM roles applied to RGs.

**Comment by aristosvo** [comment]
Hi @surlypants! Thanks for raising this. This is a well-known problem and bugfixed in 2.62.1.

I'd advise to use the bug fixed version 2.62.1 or 2.63.0.

**Comment by tombuildsstuff** [comment]
hey @surlypants 

As @aristosvo has mentioned unfortunately this was a regression in `v2.62.0` which has since been fixed in `v2.62.1` - as such you'd need to upgrade to that release (or the latest release `v2.63.0`) - but since this regressions been fixed I'm going to close this issue for the moment.

Thanks!

**Comment by surlypants** [comment]
thanks! i should've refreshed the changelog page :)
